### DISCUSSION TITLE
onClick does not work

### DISCUSSION THREAD

**Discussion by ru88s** [freshframework/fresh#716] [Q&A] [answered]
/routes/index.tsx
```js
export default function Home() {
  return (
    <button
      onClick={() => console.log("test")}
    >
      test
    </button>
  );
};
```

No log is output when I click on the "button".
Do I need to do any configuration in Deno Fresh?

**Comment by lucacasonato** [comment, marked as answer]
Click handlers and similar only work inside of island components, because only islands are shipped to the client. Please read https://fresh.deno.dev/docs/getting-started/adding-interactivity

**Reply by ru88s** [reply]
Resolved. Thank you!

**Comment by var618** [comment]
It seems I didn't read the `fresh` document carefully enough. 

I encountered the same issue today when using the onClick event on an `a` tag inside a route, trying to call openDoc (which includes a window.open call), but it just wouldn't work. 

It turns out that `island` is what handles the client-side logic.

This answer helped me, thank you!
